src/conf.py

```diff
@@ -71,84 +102,78 @@ def multiprocess_log(filename, name=__name__):
         log.addHandler(_handler)
     return log
 
-#
-#
-#
-
-SRC_DIR = os.path.dirname(inspect.getfile(inspect.currentframe())) # ll: /path/to/adaptor/src/
-PROJECT_DIR = os.path.dirname(SRC_DIR)  # ll: /path/to/adaptor/
-
-CFG_NAME = 'app.cfg'
-DYNCONFIG = configparser.SafeConfigParser(**{
-    'allow_no_value': True,
-    # these can be used like template variables
-    # https://docs.python.org/2/library/configparser.html
-    'defaults': {'dir': PROJECT_DIR}})
-DYNCONFIG.read(join(PROJECT_DIR, CFG_NAME)) # ll: /path/to/lax/app.cfg
-
-def cfg(path, default=0xDEADBEEF):
-    lu = {'True': True, 'true': True, 'False': False, 'false': False} # cast any obvious booleans
-    try:
-        val = DYNCONFIG.get(*path.split('.'))
-        return lu.get(val, val)
-    except (configparser.NoOptionError, configparser.NoSectionError): # given key in section hasn't been defined
-        if default == 0xDEADBEEF:
-            raise ValueError("no value/section set for setting at %r" % path)
-        return default
-    except Exception:
-        raise
+LOG_DIR = PROJECT_DIR
+# TODO: re-enable once changes to lax-formula are deployed
+# if ENV != DEV:
+#    LOG_DIR = '/var/log/bot-lax-adaptor/'
+utils.writable_dir(LOG_DIR)
 
 #
 #
 #
 
-ENV = cfg('general.env')
-
 PATH_TO_LAX = cfg('lax.location')
 
-CACHE_PATH = cfg('lax.cache_path')
+# TODO: re-enable once changes to lax-formula are deployed
+CACHE_PATH = join(PROJECT_DIR, 'cache')
+# if ENV != DEV:
+#    CACHE_PATH = cfg('general.cache_path', CACHE_PATH)
+
+# TODO: remove once changes to lax-formula are deployed
+CACHE_PATH = cfg('lax.cache_path', CACHE_PATH)
 
 INGEST, PUBLISH, INGEST_PUBLISH = 'ingest', 'publish', 'ingest+publish'
-INGESTED, PUBLISHED, INVALID, ERROR = 'ingested', 'published', 'invalid', 'error'
+VALIDATED, INGESTED, PUBLISHED, INVALID, ERROR = 'validated', 'ingested', 'published', 'invalid', 'error'
 
 XML_DIR = join(PROJECT_DIR, 'article-xml', 'articles')
 JSON_DIR = join(PROJECT_DIR, 'article-json')
 
-def json_load(path):
+def load(path):
     path = join(PROJECT_DIR, 'schema', path)
-    return json.load(open(path, 'r'))
+    if path.endswith('.json'):
+        return json.load(open(path, 'r'))
+    elif path.endswith('.yaml'):
+        return yaml.load(open(path, 'r'))
 
-POA_SCHEMA = json_load('api-raml/dist/model/article-poa.v1.json')
-VOR_SCHEMA = json_load('api-raml/dist/model/article-vor.v1.json')
+POA_SCHEMA = load('api-raml/dist/model/article-poa.v1.json')
+VOR_SCHEMA = load('api-raml/dist/model/article-vor.v1.json')
 
-REQUEST_SCHEMA = json_load('request-schema.json')
-RESPONSE_SCHEMA = json_load('response-schema.json')
+REQUEST_SCHEMA = load('request-schema.json')
+RESPONSE_SCHEMA = load('response-schema.json')
+
+API_SCHEMA = load('api.yaml')
+
+# can be overriden when creating an app
+API_UPLOAD_FOLDER = join(PROJECT_DIR, "uploads")
```

Comment:
yes indeed: https://github.com/elifesciences/lax-formula/blob/feat-VALIDATE/salt/lax/bot-lax-adaptor.sls#L98

and git has that folder created with a placeholder inside.

this state should be safe until the formula updates introduce the bot-lax uwsgi server, at which point the uploads and cache and log directories need to be writable by `www-data` and `elife`. I managed to remove the elife-tools `test.log` from the equation. 

the bot-lax app.cfg file: https://github.com/elifesciences/lax-formula/blob/feat-VALIDATE/salt/lax/config/opt-bot-lax-adaptor-app.cfg

I'll re-enable the upload folder to read from the cfg file, this shouldn't affect anything until the formula kicks in. the log folder and cache folder might actually be problems now that I look at them. it depends on them existing with the right permissions. might need to split the formula changes into two sets